lib/facter/framework/config/fact_groups.rb

```diff
@@ -106,12 +108,14 @@ def load_groups
 
     def ttls_to_seconds(ttls)
       duration, unit = ttls.split(' ', 2)
+      unit = '' if duration && !unit
+      unit += 's' if unit.length > 2 && unit[-1] != 's'
       seconds = STRING_TO_SECONDS[unit]
       if seconds
-        duration.to_i * seconds
+        (duration.to_i * seconds).to_i
       else
         log = Log.new(self)
-        log.error("Could not parse time unit #{unit} (try second(s), minute(s), hour(s) or day(s))")
+        log.error("Could not parse time unit #{unit} (try #{STRING_TO_SECONDS.keys})")
```

Comment:
```suggestion
        log.error("Could not parse time unit #{unit} (try #{STRING_TO_SECONDS.keys.reject(&:empty?).join(', ')})")
```

will change output from 
```
Could not parse time unit mss (try ["ns", "nanos", "nanoseconds", "us", "micros", "microseconds", "", "ms", "milis", "milliseconds", "s", "seconds", "m", "minutes", "h", "hours", "d", "days"])
```
to
```
Could not parse time unit mss (try ns, nanos, nanoseconds, us, micros, microseconds, , ms, milis, milliseconds, s, seconds, m, minutes, h, hours, d, days)
```